### PR TITLE
Adding 'otherwise' keyword.

### PR DESCRIPTION
**Pull request by kormat**
This is very unready for actual merging, etc. I mostly just want to get some feedback on the approach.

### CODE REVIEW

**Comment by kormat** [comment]
#18

**Comment by kormat** [comment]
https://gist.github.com/kormat/c8655c0ec49d2e1ffc29c2c2a1e6c694 has the test program and input that i've been using, and the one_shot_metrics output.

**Comment by jaqx0r** [comment]
Looks good so far!

Please add some more tests in the vm module: lexer_test.go turns text into a token and position stream, parser_test.go validates that the program correctly recognises input (but not the parse tree)  Nice work on compiler_test. vm_test.go tests how each instruction modifies the machine state after a single bytecode execution cycle.

The gist could be turned into an example program test through exampleProgramTests in ex_test.go.

**Comment by kormat** [comment]
`lexer_test`/`parser_test`/`vm_test`/`unparser` all updated. I'll look at `ex_test` now.

**Comment by kormat** [comment]
Hmm, i've run into a few issues:
- Matching on `/^1$/` doesn't work, as the input still contain trailing newlines. This seems like a bug rather than a feature.
- If the input file has a trailing newline, the script is executed one last time with `''` as the input, which causes the outermost `otherwise` to trigger. This seems like a related bug to the first one.
- `ex_test` doesn't like the timestamps in my `.golden`, but i can't see how the existing timestamps are allowed.

**Comment by kormat** [comment]
I've confirmed that the first two issues are actually bugs, and sent #25 to fix them.

**Comment by kormat** [comment]
Found a way to pacify `ex_test`'s passion for timestamps. (This PR now depends on #25 also).

**Comment by jaqx0r** [comment]
Wow, thanks!

File: examples/otherwise.mtail [hunk 1]
```diff
@@ -0,0 +1,18 @@
+counter yes
+counter maybe
+counter no
+
+# To make ex_test.go happy
+strptime("2016-04-25T20:14:42Z", "2006-01-02T15:04:05Z07:00")
```
Comment (jaqx0r) [inline examples/otherwise.mtail]: neat hack!

**Pull request merged**